### ISSUE TITLE
alt-tab freezes the game to a black screen

### ISSUE THREAD

**Issue by sunbeam906** [TheIndra55/TRAE-menu-hook#1]
Hi there.

I've managed to configure the source code to get it to a compilable state (maybe you want to revise the git content?). Used raw version of MinHook (all the include/src files, rather than NuGet's .lib causing detection issues -- says not found all the time). Then finally compiled and dropped in game folder as an .asi file, as per your instructions. Am using the only no-cd fix by HATRED. Game runs fine, ImGUI window shows up, F2 and F8 work to noclip and show up the mouse cursor. However.. doing alt-tab to Desktop and back will freeze the game; you can't alt-tab out, but I'm using Process Hacker replacing the default task manager, so I can kill an app by typing its process name (tra.exe;del). Just in case you wanna try.

In short.. you need to handle reset and on device lost. See: https://github.com/User1514/CSGOSimple/blob/master/CSGOSimple/Hooks.cpp#L222.

BR,
Sun

**Comment by TheIndra55** [comment]
Thank you for opening this issue, I was aware of the game freezing while alt-tab'ing out of the game. Just not something I looked into in detail as I was having way more fun reversing other stuff in the game. 

Thanks for the hint on handling the device reset that really helped, I will see if I can look into it tommorow since it's late right now.  The git repo is a mess and probably needs some revision (feel free to share your ideas on that if any, I was looking into premake too so I don't have to ship solution files).

PS I find it really cool a big modder is trying my stuff thank you for that.

**Comment by sunbeam906** [comment]
I can share more of my intentions, how I got here and what I'm currently doing. Maybe you would be interested in this as well. Though I'd recommend, if you use Discord, to add my avatar so we can talk a bit more freely than constrained by messages on github :) So: **SunBeam#5082**. I will delete this message once you add me. Thanks! P.S.: Am too in the working system, so I'd recommend up front not worrying about me nagging or bothering you :P Cheers!
